waspc/data/Generator/templates/sdk/wasp/client/router/types.ts

```diff
@@ -21,7 +21,7 @@ export type Search =
 
 type RouteDefinitionsToRoutesObj<Routes extends RoutesDefinition> = {
   [K in keyof Routes]: {
-    to: Routes[K]['to']
+    to: ExpandRouteOnOptionalStaticSegments<Routes[K]['to']>
```

Comment:
This is a bit of Typescript type level programming magic. It converts string literals that have a route with optional static segments into a union of string literals of possible route variants. 

`ExpandRouteOnOptionalStaticSegments<'/my/:id/route?'>` will result in `'/my/:id' | '/my/:id/route'` type.